ambry-server/src/test/java/com.github.ambry.server/AmbryRequestsTest.java

```diff
@@ -330,6 +331,81 @@ public void catchupStatusFailureTest() throws InterruptedException, IOException
+    PartitionId id = partitionIds.get(0);
+    int correlationId = TestUtils.RANDOM.nextInt();
+    String clientId = UtilsTest.getRandomString(10);
+    short numReplicasCaughtUpPerPartition = -1;
+    // test invalid numReplicasCaughtUpPerPartition
+    AdminRequest adminRequest = new AdminRequest(AdminRequestOrResponseType.StopBlobStore, id, correlationId, clientId);
+    StopBlobStoreAdminRequest stopBlobStoreAdminRequest =
+        new StopBlobStoreAdminRequest(numReplicasCaughtUpPerPartition, adminRequest);
+    Response response = sendRequestGetResponse(stopBlobStoreAdminRequest, ServerErrorCode.Bad_Request);
+    assertTrue("Response not of type AdminResponse", response instanceof AdminResponse);
+    // test partition unknown
+    numReplicasCaughtUpPerPartition = 3;
+    adminRequest = new AdminRequest(AdminRequestOrResponseType.StopBlobStore, null, correlationId, clientId);
+    stopBlobStoreAdminRequest = new StopBlobStoreAdminRequest(numReplicasCaughtUpPerPartition, adminRequest);
+    response = sendRequestGetResponse(stopBlobStoreAdminRequest, ServerErrorCode.Partition_Unknown);
+    assertTrue("Response not of type AdminResponse", response instanceof AdminResponse);
+    // test disable compaction failure
+    adminRequest = new AdminRequest(AdminRequestOrResponseType.StopBlobStore, id, correlationId, clientId);
+    stopBlobStoreAdminRequest = new StopBlobStoreAdminRequest(numReplicasCaughtUpPerPartition, adminRequest);
+    storageManager.returnValueOfDisablingCompaction = false;
+    response = sendRequestGetResponse(stopBlobStoreAdminRequest, ServerErrorCode.Unknown_Error);
+    assertTrue("Response not of type AdminResponse", response instanceof AdminResponse);
+    storageManager.returnValueOfDisablingCompaction = true;
+    // test disable replication failure
+    replicationManager.reset();
+    replicationManager.controlReplicationReturnVal = false;
+    adminRequest = new AdminRequest(AdminRequestOrResponseType.StopBlobStore, id, correlationId, clientId);
+    stopBlobStoreAdminRequest = new StopBlobStoreAdminRequest(numReplicasCaughtUpPerPartition, adminRequest);
+    response = sendRequestGetResponse(stopBlobStoreAdminRequest, ServerErrorCode.Bad_Request);
+    assertTrue("Response not of type AdminResponse", response instanceof AdminResponse);
+    // test peers catchup failure
+    replicationManager.reset();
+    replicationManager.controlReplicationReturnVal = true;
+    // all replicas of this partition > acceptableLag
+    generateLagOverrides(1, 1);
+    adminRequest = new AdminRequest(AdminRequestOrResponseType.StopBlobStore, id, correlationId, clientId);
+    stopBlobStoreAdminRequest = new StopBlobStoreAdminRequest(numReplicasCaughtUpPerPartition, adminRequest);
+    response = sendRequestGetResponse(stopBlobStoreAdminRequest, ServerErrorCode.Catchup_Unfinished);
+    assertTrue("Response not of type AdminResponse", response instanceof AdminResponse);
```

Comment:
dont you have to check the response for the error code? (in all the tests above)

Reply:
`sendRequestGetResponse(stopBlobStoreAdminRequest, ServerErrorCode.Catchup_Unfinished);` the check for response error code is covered in this method.  "ServerErrorCode.Catchup_Unfinished" for example.

Reply:
ah ok - that covers the check i wanted